modules/post/windows/manage/reflective_dll_inject.rb

```diff
@@ -6,45 +6,187 @@
-      'Name'          => 'Windows Manage Reflective DLL Injection Module',
-      'Description'   => %q{
-        This module will inject into the memory of a process a specified Reflective DLL.
-      },
-      'License'      => MSF_LICENSE,
-      'Author'       => [ 'Ben Campbell'],
-      'Platform'     => [ 'win' ],
-      'SessionTypes' => [ 'meterpreter' ],
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'Windows Manage Reflective DLL Injection
+      Module with arguments',
+      'Description' => '
+      This module will inject a specified reflective DLL
+      into the memory of a process, new or existing,
+      passing any arguments.
+      ',
+      'License' => MSF_LICENSE,
+      'Author' => 'Ben Campbell,b4rtik',
+      'Platform' => 'win',
+      'SessionTypes' => ['meterpreter'],
       'References'   =>
-        [
-          [ 'URL', 'https://github.com/stephenfewer/ReflectiveDLLInjection' ]
-        ]
-    ))
-
+      [
+        [ 'URL', 'https://github.com/stephenfewer/ReflectiveDLLInjection' ]
+      ]))
     register_options(
       [
-        OptPath.new('PATH',[true, 'Reflective DLL to inject into memory of a process.']),
-        OptInt.new('PID',[true, 'Process Identifier to inject of process to inject payload.']),
-      ])
+        OptPath.new('PATH', [true, 'Reflective DLL to inject into memory of a process.']),
+        OptInt.new('PID', [false, 'Pid  to inject', 0]),
+        OptString.new('PROCESS', [false, 'Process to spawn','notepad.exe']),
+        OptString.new('ARGUMENTS', [false, 'Command line arguments']),
+        OptInt.new('WAIT', [false, 'Time in seconds to wait.', 0])
+      ], self.class
+    )
+
+    register_advanced_options(
+      [
+        OptBool.new('KILL',   [ true, 'Kill the injected process at the end of the task. ', false ])
+      ]
+    )
   end
 
-  # Run Method for when run command is issued
   def run
-    # syinfo is only on meterpreter sessions
-    print_status("Running module against #{sysinfo['Computer']}") if not sysinfo.nil?
+    dll_path = gen_dll_path
+    if File.file?(dll_path)
+      run_dll(dll_path)
+    else
+      print_bad("Dll not found #{dll_path}")
+    end
+  end
+
+  def gen_dll_path
+    dll_path = ::File.expand_path(datastore['PATH'])
+    dll_path
+  end
+
+  def sanitize_process_name(process_name)
+    if process_name.split(//).last(4).join.eql? '.exe'
+      process_name
+    else
+      process_name + '.exe'
+    end
+  end
+
+  def pid_exists(pid)
+    mypid = client.sys.process.getpid.to_i
+
+    if pid == mypid
+      print_bad("Invalid PID")
+      return false
+    end
+
+    host_processes = client.sys.process.get_processes
+    if host_processes.length < 1
+      print_bad("No running processes found on the target host.")
+      return false
+    end
+
+    theprocess = host_processes.find {|x| x["pid"] == pid}
+
+    if ( theprocess.nil? )
+      return false
+    else
+      return true
+    end
```

Comment:
You could combine all 5 of these lines into simply:

```suggestion
    !theprocess.nil?
```

You also don't need the explicit `return` in ruby. The last expression of the function will be returned to the caller which is why in the suggestion the `return` keyword is missing.